protocol/proofs/new-chains.md

```diff
@@ -0,0 +1,70 @@
+`OptimismPortal`. The fault proof system creates two cyclic dependencies:
+
+1. The `faultGameGenesisOutputRoot` is used by the L1 `AnchorStateRegistry` contract as the initial anchor state for
+   permissioned and permissionless games. For new chains it must be set to the L2 genesis output root, which is
+   unknowable prior to deploing the L1 contracts.
+2. The `faultGameAbsolutePrestate` is used by the L1 `FaultDisputeGame` contract to define the version of `op-program`
+   to be used as part of permissioned and permissionless dispute games. `op-program` requires the L2 chain
+   configuration (rollup config and L2 genesis state) to be available in the version of the superchain-registry that it
+   includes to operate correctly. Similar to above, the L2 chain configuration is unknowable prior to deploying the L1
+   contracts. Additionally, chains cannot be added to the superchain-registry until _after_ they have launched.
+
+# Proposed Solution
+
+TBD
+
+# Alternatives Considered
+
+## Defer Deployment of Dispute Game Types
+
+Remove the `FaultDisputeGame` and its dependent contracts (including `AnchorStateRegistry`) from the initial L1
+deployment. The `DisputeGameFactory` would still be deployment, but with no supported game types. It would be impossible
+to create any dispute games and thus withdrawals would be disabled. The `faultGameGenesisOutputRoot` and
+`faultGameAbsolutePrestate` values would not be required ahead of this initial L1 deployment as they are only used by
+the dispute game implementations.
+
+The L2 genesis state could then be created based on the deployed L1 contracts. Optionally the chain could be started at
+this point without support for withdrawals.
+
+The config would need to be added to a version of `op-program` either by adding the chain to the superchain-registry,
+updating the superchain-registry dependency in the optimism monorepo and tagging a new release of `op-program` or by
+creating some kind of custom build of `op-program`.
+
+Finally, the `ProxyAdminOwner` would deploy the `FaultDisputeGame` and dependent contracts, providing the correct
+`faultGameGenesisOutputRoot` and `faultGameAbsolutePrestate` values and call `setImplementation` on the
+`DisputeGameFactory`. Withdrawals would then be possible.
+
+### Pros
+
+* Neatly avoids the cyclic dependency.
+* Chains that are not supporting permissionless games yet could deploy only the permissioned game, preventing creating
+  permissionless games entirely.
+
+### Cons
+
+* The `ProxyAdminOwner` is required to perform the `setImplementation` calls because ownership of the L1 contracts has
+  already been transferred. In the current system, the L1 contract deployment can be performed by anyone and the system
+  is fully configured (albeit incorrectly for proofs) before ownership is transferred.
+* It remains unclear at what point a chain can be added to the superchain-registry or how a version of `op-program` that
+  supports the new chain would be created.
```

Comment:
@geoknee This is why you got tagged in as a reviewer. Things are simpler for `op-program` if the config is always coming from the superchain-registry but I suspect there's a bunch of checks in the superchain-registry that could be affected by not yet having the dispute game implementations deployed and in general it seems awkward to have adding the chain to the superchain registry on the critical path for deploying a new chain.

We have similar weirdness with op-challenger, op-node, op-geth etc which pull config from the superchain-registry as new chains have to configure everything the hard way by specifying the rollup config/genesis manually and then later switch to using `--network` once the chain is in the superchain-registry which introduces a lot of potential for errors and winding up with mismatched configs. But at least for those it's doable.

Reply:
The registry does support adding a chain as a "standard candidate" chain, which will later become a fully standard chain (in fact this is the only way to add a chain at present). This means that when a chain is added, not all of the standard checks run on it (yet). So adding the chain to the registry is likely not going to be blocked on having the chain fully deployed and set up. Only "promoting the chain to standard" would be blocked, this is just a small PR bumping the superchain_level and potentially adding a couple of contract addresses.

I agree it is awkward to have the dependency the other way around (blocking chain deployment on adding the chain to the registry). OTOH maybe it is reasonable, for chains opting into a shared governance system with coordinated hardforks and so on? If we leaned into this, it might help i) adoption of the `--network` flags (these are not widely used) and ii) reduce the chance for mismatched configs in the end.